app/core/caching.py

```
@@ -0,0 +1,99 @@
+def invalidate_cache(key_prefix: str):
+    """
+    Decorator to invalidate cache after write operations
+    
+    Args:
+        key_prefix: Prefix for the cache key to invalidate
+    """
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        async def wrapper(*args, **kwargs):
+            # Execute the function first
+            result = await func(*args, **kwargs) if hasattr(func, '__call__') else func(*args, **kwargs)
```

Ruff (0.13.1)

Using `hasattr(x, "__call__")` to test if x is callable is unreliable. Use `callable(x)` for consistent results.

Replace with `callable()`

(B004)